File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/models/Locatie.kt
package nl.bsodetheepot.mobile.data.models

import io.github.jan.supabase.postgrest.postgrest
import io.github.jan.supabase.postgrest.query.Order
import kotlinx.serialization.Serializable
import nl.bsodetheepot.mobile.data.services.SupabaseManager

/**
 * Tabel `kasboek_locaties` — gedeelde locatielijst, hergebruikt door kasboek,
 * maaltijdlijst, prikbord en weekplanningen (net als in de webapp).
 */
@Serializable
data class Locatie(
    val id: String,
    val naam: String,
    val actief: Boolean,
)

object LocatieService {
    suspend fun actieveLocaties(): List<Locatie> =
        SupabaseManager.client.postgrest["kasboek_locaties"]
            .select {
                filter { eq("actief", true) }
                order("naam", Order.ASCENDING)
            }
            .decodeList()
}

File: TheepotMobile/TheepotMobile/Models/Locatie.swift
import Foundation

/// Tabel `kasboek_locaties` — gedeelde locatielijst, hergebruikt door kasboek,
/// maaltijdlijst, prikbord en weekplanningen (net als in de webapp).
struct Locatie: Codable, Identifiable, Hashable {
    let id: String
    let naam: String
    let actief: Bool
}

enum LocatieService {
    static func actieveLocaties() async throws -> [Locatie] {
        try await SupabaseManager.client
            .from("kasboek_locaties")
            .select()
            .eq("actief", value: true)
            .order("naam")
            .execute()
            .value
    }
}
